lib/http/2/connection.rb

```diff
@@ -446,12 +448,15 @@ def connection_management(frame)
           # the connection, although a new connection can be established
           # for new streams.
           @state = :closed
+          @closed_since = Time.now
           emit(:goaway, frame[:last_stream], frame[:error], frame[:payload])
         when :altsvc, :blocked
           emit(frame[:type], frame)
         else
           connection_error
         end
+      when :closed
+        connection_error if (Time.now - @closed_since) > 15
```

Comment:
Overall, what you wired up here makes sense. 👍 

One question I have is.. Current logic will cover {settings, window_update, ping, goaway} connection management frames. What about other types of frames, do we need to do anything there? Per https://github.com/igrigorik/http-2/issues/139#issuecomment-414138927...

> After sending a GOAWAY frame, the sender can discard frames for
> streams initiated by the receiver with identifiers higher than the
> identified last stream. However, any frames that alter connection
> state cannot be completely ignored. For instance, HEADERS,
> PUSH_PROMISE, and CONTINUATION frames MUST be minimally processed to
> ensure the state maintained for header compression is consistent (see
> Section 4.3); similarly, DATA frames MUST be counted toward the
> connection flow-control window. Failure to process these frames can
> cause flow control or header compression state to become
> unsynchronized.

Anything else / special we need to think about with respect to the above?

Reply:
I'd say it's already covered [here](https://github.com/igrigorik/http-2/blob/3f0cf83c68d11cd04300f186007a28b132b7f8f4/lib/http/2/connection.rb#L253), [here](https://github.com/igrigorik/http-2/blob/3f0cf83c68d11cd04300f186007a28b132b7f8f4/lib/http/2/connection.rb#L253) or [here](https://github.com/igrigorik/http-2/blob/3f0cf83c68d11cd04300f186007a28b132b7f8f4/lib/http/2/connection.rb#L321-L324).

Reply:
Fair enough. One corner case here is that we'll be ignoring _all_ frames, irrespective of their ID being established before or after GOAWAY is emitted, but that's probably OK. If that surfaces as an issue, we can revisit.

Reply:
That is true, and I thought about that. One should throw protocol error for when a stream id is higher than the last stream id from the goaway frame. But I thought that was out of scope. Am I wrong?

Reply:
I think it's worth leaving a comment in the source, stating that we're _not_ enforcing this behavior. I don't think we need to block this PR on implementing that though.

Reply:
after reading better, I think I advised for something which is not RFC-compliant. I think this PR covers everything necessary.